### ISSUE TITLE
Using primitive types as Payload causes compile time error

### ISSUE THREAD

**Issue by arangamani** [goadesign/goa#304]
When using primitive types such as `String`, `Any`, etc. in the payload for an action, the generated code in `app/controllers.go` throws a compile time error.
Resource Definition:
```go
var _ = Resource("my_resource", func() {
	Action("do", func() {
		Routing(
			POST("/:id/actions/do"),
		)
		Description("Do something")
		Params(func() {
			Param("id", String, "The ID")
		})
		Payload(Any)
		Response(Accepted)
		Response(NotFound)
		Response(BadRequest)
	})
})
```

Generated code in `app/contexts.go`:
```go
type DoMyResourcePayload interface{}
```

Generated code in `app/controllers.go`:
```go
func unmarshalDoMyResourcePayload(ctx context.Context, req *http.Request) error {
	payload := DoMyResourcePayload{} // <<<- error
	if err := goa.RequestService(ctx).DecodeRequest(req, payload); err != nil {
		return err
	}
	goa.Request(ctx).Payload = payload
	return nil
}

```

**Comment by raphael** [comment]
Thanks, this is actually a feature request. There should be an error message though instead of incorrect behavior... #132 is also asking for support of primitive types when defining response media types.

**Comment by raphael** [comment]
Fixed by #309
